packages/react-native/android/src/main/java/com/dashx/sdk/DashXClient.kt

```diff
@@ -194,6 +196,55 @@ class DashXClient private constructor() {
         })
     }
 
+    fun trackAppStarted() {
+        val context = reactApplicationContext?.applicationContext ?: return
+
+        val packageInfo = getPackageInfo(context)
+        val currentBuild = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            packageInfo.longVersionCode
+        } else {
+            packageInfo.versionCode.toLong()
+        }
+
+        val eventProperties = Arguments.createMap()
+        eventProperties.putString("version", packageInfo.versionName)
+        eventProperties.putString("build", currentBuild.toString())
+
+        when {
+            getDashXSharedPreferences(context).getLong(SHARED_PREFERENCES_KEY_BUILD, Long.MIN_VALUE) == Long.MIN_VALUE
+                -> track(INTERNAL_EVENT_APP_INSTALLED, eventProperties)
+            getDashXSharedPreferences(context).getLong(SHARED_PREFERENCES_KEY_BUILD, Long.MIN_VALUE) < currentBuild
+                -> track(INTERNAL_EVENT_APP_UPDATED, eventProperties)
+            else -> track(INTERNAL_EVENT_APP_OPENED, eventProperties)
+        }
+
+        val editor: SharedPreferences.Editor = getDashXSharedPreferences(context).edit()
+        editor.putString(SHARED_PREFERENCES_KEY_VERSION, packageInfo.versionName)
+        editor.putLong(SHARED_PREFERENCES_KEY_BUILD, currentBuild)
```

Comment:
Could we avoid running these 2 lines in the Application Opened case?

Reply:
We need to run these to track if App is updated, We can run these here or when `DashXClient` is initialized, I put these here because I don't extra keys in SharedPreferences  if `trackAppLifecyle` is off.

Reply:
I agree. But there are 3 cases above (installed / updated / opened). You only need to run this code in the first 2 cases I believe. If you don't need to modify SharedPreferences, we shouldn't:

```
Note: This class provides strong consistency guarantees. It is using expensive operations which might slow down an app. Frequently changing properties or properties where loss can be tolerated should use other mechanisms. For more details read the comments on Editor#commit() and Editor#apply().
```

Since the process involves writing to the disk, even if it's async, I'm wondering if setting a simple flag `shouldEditPreferences = true` at the beginning, and then setting it to `false` for the OPENED case makes the app bootup time a tiny bit less for a tiny bit more code. :)